ProjectLighthouse/Controllers/ReviewController.cs

```diff
@@ -74,5 +81,177 @@ public async Task<IActionResult> DPadRate(int slotId, [FromQuery] int rating)
+
+            await this.database.SaveChangesAsync();
+
+            return this.Ok();
+        }
+
+        [HttpGet("reviewsFor/user/{slotId:int}")]
+        public async Task<IActionResult> ReviewsFor(int slotId, [FromQuery] int pageStart = 1, [FromQuery] int pageSize = 10) 
+        {
+            User? user = await this.database.UserFromRequest(this.Request);
+            if (user == null) return this.StatusCode(403, "");
+
+            Token? token = await this.database.TokenFromRequest(this.Request);
+            if (token == null) return this.StatusCode(403, "");
+
+            GameVersion gameVersion = token.GameVersion;
+
+            Random rand = new();
+
+            IEnumerable<Review> reviews = this.database.Reviews.Where(r => r.SlotId == slotId && r.Slot.GameVersion <= gameVersion)
+                .Include(r => r.Reviewer)
+                .Include(r => r.Slot)
+                .AsEnumerable() // performance? Needed for next line (ThumbsUp is not in DB)
+                .OrderByDescending(r => r.ThumbsUp)
+                .ThenByDescending(_ => rand.Next())
+                .Skip(pageStart - 1)
+                .Take(pageSize);
+                
+            string inner = Enumerable.Aggregate(reviews, string.Empty, (current, review) => {
+                RatedLevel? ratedLevel = this.database.RatedLevels.FirstOrDefault(r => r.SlotId == slotId && r.UserId == review.ReviewerId);
+                RatedReview? ratedReview = this.database.RatedReviews.FirstOrDefault(r => r.ReviewId == review.ReviewId && r.UserId == user.UserId);
+                
+                return current + review.Serialize(ratedLevel, ratedReview);
+            });
+
+            string response = LbpSerializer.TaggedStringElement("reviews", inner, new Dictionary<string, object>
+                    {
+                        {
+                            "hint_start", pageStart + pageSize
+                        },
+                        {
+                            "hint", pageStart // not sure
+                        },
+                    });
+            return this.Ok(response);
+        }
+
+        [HttpGet("reviewsBy/{username}")]
+        public async Task<IActionResult> ReviewsBy(string username, [FromQuery] int pageStart = 1, [FromQuery] int pageSize = 10) 
+        {
+            User? user = await this.database.UserFromRequest(this.Request);
+            if (user == null) return this.StatusCode(403, "");
+
+            Token? token = await this.database.TokenFromRequest(this.Request);
+            if (token == null) return this.StatusCode(403, "");
+
+            GameVersion gameVersion = token.GameVersion;
+
+            IEnumerable<Review> reviews = this.database.Reviews.Where(r => r.Reviewer.Username == username && r.Slot.GameVersion <= gameVersion)
+                .Include(r => r.Reviewer)
+                .Include(r => r.Slot)
+                .AsEnumerable() // performance?
+                .OrderByDescending(r => r.Timestamp)
```

Comment:
Why cant this be ordered server-side?

Reply:
You're right, I think it came from something I copied and pasted. Fixed https://github.com/LBPUnion/project-lighthouse/pull/62/commits/abfb1c4cd86f0a222ad14f148d7a1e9931e83444